data-prepper-expression/src/main/java/org/opensearch/dataprepper/expression/OperatorConfiguration.java

```diff
@@ -29,37 +29,83 @@ public NumericCompareOperator greaterThanOperator() {
         final Map<Class<? extends Number>, BiFunction<Object, Object, Boolean>> intOperations =
                 Map.of(
                         Integer.class, (lhs, rhs) -> (Integer) lhs > (Integer) rhs,
-                        Float.class, (lhs, rhs) -> (Integer) lhs > (Float) rhs
+                        Float.class, (lhs, rhs) -> (Integer) lhs > (Float) rhs,
+                        Long.class, (lhs, rhs) -> (Integer) lhs > (Long) rhs,
+                        Double.class, (lhs, rhs) -> (Integer) lhs > (Double) rhs
                 );
         final Map<Class<? extends Number>, BiFunction<Object, Object, Boolean>> floatOperations =
                 Map.of(
                     Integer.class, (lhs, rhs) -> (Float) lhs > (Integer) rhs,
-                    Float.class, (lhs, rhs) -> (Float) lhs > (Float) rhs
+                    Float.class, (lhs, rhs) -> (Float) lhs > (Float) rhs,
+                    Long.class, (lhs, rhs) -> (Float) lhs > (Long) rhs,
+                    Double.class, (lhs, rhs) -> (Float) lhs > (Double) rhs
+                );
+
+        final Map<Class<? extends Number>, BiFunction<Object, Object, Boolean>> longOperations =
+                Map.of(
+                    Integer.class, (lhs, rhs) -> (Long) lhs > (Integer) rhs,
+                    Float.class, (lhs, rhs) -> (Long) lhs > (Float)rhs,
+                    Long.class, (lhs, rhs) -> (Long) lhs > (Long) rhs,
+                    Double.class, (lhs, rhs) -> (Long) lhs > (Double)rhs
+                );
+
+        final Map<Class<? extends Number>, BiFunction<Object, Object, Boolean>> doubleOperations =
```

Comment:
Seems like there could be a less manual way to create these combinations

Reply:
It may be possible. But that's something we should try as a separate issue. Given the urgency to fix this issue, I took this approach.